pkgs/development/libraries/crypto++/default.nix

```diff
@@ -8,7 +8,9 @@ stdenv.mkDerivation rec {
     sha256 = "0x1mqpz1v071cfrw4grbw7z734cxnpry1qh2b6rsmcx6nkyd5gsw";
   };
 
-  patches = stdenv.lib.optional (stdenv.system != "i686-cygwin") ./dll.patch;
+  patches = [ (stdenv.lib.optional (stdenv.system != "i686-cygwin") ./dll.patch)
```

Comment:
This is going to cause a mass-rebuild on linux because patches will evaluate to `[ ./dll.patch [] ]` instead of `[ ./dll.patch ]`. Can you try

```
 (stdenv.lib.optional (stdenv.system != "i686-cygwin") ./dll.patch) ++ (stdenv.lib.optional stdenv.isDarwin ./GNUmakefile.patch)
```

instead?